numpy/core/tests/test_multiarray.py

```diff
@@ -3548,21 +3566,55 @@ def test_test_zero_rank(self):
 
 
 class TestPickling(object):
+    def test_highest_available_pickle_protocol(self):
+        try:
+            import pickle5
+        except ImportError:
+            pickle5 = None
+
+        if sys.version_info[:2] >= (3, 8) or pickle5 is not None:
+            assert pickle.HIGHEST_PROTOCOL >= 5
+        else:
+            assert pickle.HIGHEST_PROTOCOL < 5
+
+    def test_array_with_dtype_has_object_pickling(self):
```

Comment:
I would rename this to `test_record_array_with_object_dtype`. There is already "pickling" in the test class name.